File: 1_4/1task.kt
fun main() {
    val N = 4
    if (N > 26) {
        println("Incorrect data")
    } 
    else {
        for (i in N downTo 1) {
            for (k in N downTo i) {
                val Str = 'A' + k - 1
                print("$Str ")
            }
            println()
        }
    }
}

File: 1_4/1task.rb
N = 4
if N > 26
    puts "Incorrect data"
else
    for i in (N).downto(1)
        for k in (N).downto(i)
            Str = ('A'.ord + k - 1).chr
            print Str + " "
        end
        puts
    end
end